### ISSUE TITLE
module 'unittest.mock' has no attribute '__version__'

### ISSUE THREAD

**Issue by jorti** [torproject/stem#56]
With Python 3.9.0a3, the tests fail to run with this error:

~~~
+ /usr/bin/python3 run_tests.py --unit
======================================================================
                             INITIALISING                             
======================================================================
  stem version...                            1.8.0
  python version...                          3.9.0a3
  operating system...                        Linux
  cryptography version...                    missing
  mock version...                            failed
module 'unittest.mock' has no attribute '__version__'
~~~

It looks like `__version__` was removed:
https://github.com/python/cpython/pull/17977

Originally reported downstream in Fedora:
https://bugzilla.redhat.com/show_bug.cgi?id=1797690

**Comment by atagar** [comment]
Thanks Juan, thanks Miro! Replied on that ticket.

I'll remove that code in a bit. Mock's version is moot for Stem 2.x because we drop Python 2.x support (and as such mock is a builtin).

The bigger question is: should we hotfix this into Stem 1.8? Python shouldn't break backward compatibility during minor version bumps so I'd prefer if they fix this on their end, but if that's a no-go I suppose we can go the hotfix route.

**Comment by encukou** [comment]
Python 3.9 is in alpha; 3.9.0 final won't be released not coming out until [October](https://www.python.org/dev/peps/pep-0596/). I don't think you need a hotfix.

**Comment by atagar** [comment]
Just remembered that we already dropped mock's version check from our master branch...

https://gitweb.torproject.org/stem.git/commit/?id=5691ff0

Feel free to reopen if this misses anything.

**Comment by eli-schwartz** [comment]
> The bigger question is: should we hotfix this into Stem 1.8? Python shouldn't break backward compatibility during minor version bumps so I'd prefer if they fix this on their end, but if that's a no-go I suppose we can go the hotfix route.

> Python 3.9 is in alpha; 3.9.0 final won't be released until [October](https://www.python.org/dev/peps/pep-0596/#schedule). I don't think you need a hotfix.


Given CPython has no intent to reintroduce this, I'd appreciate a hotfix. Currently, this module cannot be successfully rebuilt with enabled tests on python 3.9, and it's one of the blockers for Arch Linux to stabilize on python 3.9.

Yes, ideally you could just say "by the time python 3.9 is released in October there will be stem 2.0", but October is here and gone, and still no stem 2.0, so in the end it turns out to be a problem after all.
